content/realtime/metachannels.textile

```diff
@@ -115,17 +116,150 @@ metaChannel.subscribe('channel.region.inactive', (msg) => {
+
+Sampled connection and request events can be used to observe the usage of Ably services. Events are published to metachannels that contain metadata for a sample of connections and API requests. These events can be used to compile statistics on app usage, enabling you to perform arbitrary data processing and aggregation related to client population and client activity.
+
+Note that this is an Enterprise-only feature and must be enabled for each app you would like to sample. Please "contact us":https://ably.com/contact to enable this functionality.
+
+Events are published on the metachannels *@[meta]clientEvents:connections@* and *@[meta]clientEvents:apiRequests@*.
+
+The following code snippet is an example of subscribing to connection events on the @[meta]clientEvents:connections@ channel:
+
+```[javascript]
+const channel = ably.channels.get('[meta]clientEvents:connections');
+
+channel.subscribe('connection.opened', (msg) => {
+    console.log('connection opened: ', msg.data);
+})
+
+channel.subscribe('connection.closed', (msg) => {
+    console.log('connection closed: ', msg.data);
+})
+
+channel.subscribe('connection.failed', (msg) => {
+    console.log('connection failed: ', msg.data);
+})
+
+channel.subscribe('connection.refused', (msg) => {
+    console.log('connection refused: ', msg.data);
+})
+```
+
+h3(#sample). Sample rate
+
+The rate of sampling dictates the number of events published to the metachannels. For example, a rate of 0.1% for connections would publish every 1000th connection event. The sample rate can be configured independently for connections and API requests within the same app.
+
+h3(#connections). Connections
+
+Sampled connection events for an app are published to the channel @[meta]clientEvents:connections@. Connection events can be used to compile statistics relating to client population.
+
+The connection event types are:
+
+- connection.opened := The connection was successfully made by the client to Ably.
+- connection.closed := The connection was explicitly closed by the client.
+- connection.refused := The connection was rejected for an expected reason, such as invalid credentials or a malformed request.
+- connection.failed :=  The connection was rejected for an unexpected reason, such as a network failure.
+
+Events contain the following metadata in the @data@ field:
+
+- name := The event type, for example @connection.opened@.<br>__Type: @String@__
+- time := The time the connection was attempted, as milliseconds since the epoch.<br>__Type: @Integer@__
+- host := The host the connection was made to.<br>__Type: @String@__
+- clientId := The ID of the client that attempted the connection.<br>__Type: @String@__
+- requestId := The unique ID of the connection.<br>__Type: @String@__
+- region := The region the connection was made from.<br>__Type: @String@__
+- headers := The headers sent with the connection.<br>__Type: @JSON Object@__
+- query := The details of the connection query.<br>__Type: @JSON Object@__
```

Comment:
this could be pretty explicit, eg "The parsed query string of the connect request, redact to remove auth params, containing connection information sent by the client library such as the library version and any custom transport parameters"

Reply:
Updated based on this conversation.